### PR TITLE
Properly handle namespaces for enum configuration options

### LINKED ISSUE #1125: Inconsistent naming scheme for whitelisting / blacklisting vs. rustified enum configurations.
The configuration for rustified-enum uses `canonical_name`. The configuration for whitelisting uses `canonical_path`.

I think rustified_enum and similar configs should use `canonical_path` too, since that handles namespaces properly.

### PR DESCRIPTION
**Pull request by tamird**
...by using canonical_path rather than canonical_name.

Fixes #1125.

r? @emilio

### CODE REVIEW

**Comment by fitzgen** [comment]
Can you add a note to the unreleased section of the changelog about this breaking change? Thanks!

**Comment by fitzgen** [comment]
And did it fix the issues you were seeing in that other bug?

**Comment by tamird** [comment]
Added a note to the changelog. I haven't had a chance to test the other issue yet, I may get to it today (but may not).

File: CHANGELOG.md [hunk 1]
```diff
@@ -39,7 +39,13 @@ Released YYYY/MM/DD
...
 ## Changed
 
-* TODO (or remove section if none)
+* The `bindgen::Builder::{constified_enum_module,{bitfield,rustified}_enum}`
+  builder methods and their corresponding CLI flags now compare their argument
+  to the C/C++ `enum`'s "canonical path", which includes leading namespaces,
+  rather than its "canonical name", which does not. This is a breaking change
+  that requires callers which target a namespaced C++ enum to call e.g.
+  `bitfield_enum("<namespace>::<enum_name>")` rather than e.g.
+  `bitfield_enum("<enum_name>")`. [#1162][]
```
Comment (fitzgen) [inline CHANGELOG.md]: Thanks!

**Pull request merged**